src/validators/signing/tests/test_signing.py

```diff
@@ -0,0 +1,157 @@
+
+
+def _mocked_remote_signer(url, **kwargs) -> CallbackResult:
+    public_key_to_sign_for = BLSPubkey(bytes.fromhex(url.path.split('/')[-1][2:]))
+
+    try:
+        corresponding_private_key = _REMOTE_SIGNER_PUBKEY_PRIVKEY_MAPPING[public_key_to_sign_for]
+    except KeyError:
+        return CallbackResult(status=404, body='Not Found')
+
+    signature = bls.Sign(corresponding_private_key, bytes.fromhex(kwargs['json']['signingRoot']))
+
+    return CallbackResult(payload={'signature': f'0x{signature.hex()}'})
+
+
+class TestSigning:
+    @pytest.mark.usefixtures('fake_settings')
+    @pytest.mark.parametrize(
+        ['key_location'],
+        [
+            pytest.param('local'),
+            pytest.param('remote'),
+        ],
+    )
+    @pytest.mark.parametrize(
+        ['_mocked_oracle_committee'],
+        [
+            pytest.param((1, 1), id='Single oracle'),
+            pytest.param((10, 5), id='10 oracles with recovery threshold of 5'),
+            pytest.param((10, 10), id='10 oracles with recovery threshold of 10'),
+        ],
+        indirect=True,
+    )
+    async def test_get_exit_signature_shards(
+        self,
+        key_location: str,
+        create_validator_keypair: Callable,
+        fork: ConsensusFork,
+        remote_signer_url: str,
+        mocked_get_oracles: Callable,
+        _mocked_oracle_committee: OracleCommittee,
+    ):
+        oracles = await mocked_get_oracles()
+
+        validator_privkey, validator_pubkey = create_validator_keypair()
+        validator_index = 123
+
+        if key_location == 'local':
+            shards = get_exit_signature_shards(
+                validator_index=validator_index,
+                private_key=validator_privkey,
+                oracles=oracles,
+                fork=fork,
+            )
+        else:
+            settings.remote_signer_url = remote_signer_url
+
+            private_key_shares = private_key_to_private_key_shares(
+                private_key=validator_privkey,
+                threshold=oracles.exit_signature_recover_threshold,
+                total=len(oracles.public_keys),
+            )
+
+            global _REMOTE_SIGNER_PUBKEY_PRIVKEY_MAPPING
+            _REMOTE_SIGNER_PUBKEY_PRIVKEY_MAPPING = {
+                bls.SkToPk(pk_share): pk_share for pk_share in private_key_shares
+            }
+
+            with aioresponses() as m:
+                m.post(
+                    re.compile(f'^{settings.remote_signer_url}/api/v1/eth2/sign/\\w{{98}}$'),
+                    callback=_mocked_remote_signer,
+                    repeat=True,
+                )
+                shards = await get_exit_signature_shards_remote_signer(
+                    validator_index=validator_index,
+                    validator_pubkey_shares=[
+                        bls.SkToPk(priv_key) for priv_key in private_key_shares
+                    ],
+                    oracles=oracles,
+                    fork=fork,
+                )
```

Comment:
Can you reuse `_mocked_remote_signer` here? Looks almost identical.

Reply:
This one was a bit harder, the tests under `src/validators/signing/tests/test_signing.py` were more unit-testy and isolated.

But I have deduplicated the code, the only downside to it is that those tests now take slightly longer to run because of a longer setup time. They now depend on the `_generate_key_shares` fixture which takes a while to set up.